src/qt/trafficgraphwidget.cpp

```diff
@@ -103,74 +168,90 @@ void TrafficGraphWidget::paintEvent(QPaintEvent *)
         }
     }
 
-    if(!vSamplesIn.empty()) {
-        QPainterPath p;
-        paintPath(p, vSamplesIn);
-        painter.fillPath(p, QColor(0, 255, 0, 128));
-        painter.setPen(Qt::green);
-        painter.drawPath(p);
-    }
-    if(!vSamplesOut.empty()) {
-        QPainterPath p;
-        paintPath(p, vSamplesOut);
-        painter.fillPath(p, QColor(255, 0, 0, 128));
-        painter.setPen(Qt::red);
-        painter.drawPath(p);
-    }
+	//MLUMIN 05_22: Paint path for visual samples 
+    QPainterPath samplesPathIn;
+    paintPath(samplesPathIn, vSamplesToDisplayIn);
+    painter.fillPath(samplesPathIn, QColor(0, 255, 0, 128));
+    painter.setPen(Qt::green);
+    painter.drawPath(samplesPathIn);
+
+    QPainterPath samplesPathOut;
+    paintPath(samplesPathOut, vSamplesToDisplayOut);
+    painter.fillPath(samplesPathOut, QColor(255, 0, 0, 128));
+    painter.setPen(Qt::red);
+    painter.drawPath(samplesPathOut);
 }
 
 void TrafficGraphWidget::updateRates()
 {
-    if(!clientModel) return;
-
-    quint64 bytesIn = clientModel->getTotalBytesRecv(),
-            bytesOut = clientModel->getTotalBytesSent();
-    float inRate = (bytesIn - nLastBytesIn) / 1024.0f * 1000 / timer->interval();
-    float outRate = (bytesOut - nLastBytesOut) / 1024.0f * 1000 / timer->interval();
-    vSamplesIn.push_front(inRate);
-    vSamplesOut.push_front(outRate);
+    if (!clientModel) {
+        return;
+    }
+
+    quint64 bytesIn = clientModel->getTotalBytesRecv();
+    quint64 bytesOut = clientModel->getTotalBytesSent();
+
+    float inRate = (bytesIn - nLastBytesIn) / 1024.0f * 1000.0f / timer->interval();
+    float outRate = (bytesOut - nLastBytesOut) / 1024.0f * 1000.0f / timer->interval();
```

Comment:
Nit: while we're at it, most other features in the UI use `10^3` as kilo instead of `2^10`, so while we're changing this, we may want to make that consistent?

Reply:
Sure. Consistency is good.